File: javascript/001_fibonacci_sequence_test.js
var assert = require('assert');
var fibonacci = require('./001_fibonacci_sequence.js');

// zero to one
assert.equal(0, fibonacci.calculate(0));
assert.equal(1, fibonacci.calculate(1));

// two to four
assert.equal(1, fibonacci.calculate(2));
assert.equal(2, fibonacci.calculate(3));
assert.equal(3, fibonacci.calculate(4));

// large
assert.equal(5, fibonacci.calculate(5));
assert.equal(144, fibonacci.calculate(12));

// negative one to negative four
assert.equal(1, fibonacci.calculate(-1));
assert.equal(-1, fibonacci.calculate(-2));
assert.equal(2, fibonacci.calculate(-3));
assert.equal(-3, fibonacci.calculate(-4));

File: c/001_fibonacci_sequence_test.c
#include <assert.h>
#include "001_fibonacci_sequence.h"

int main(void)
{
  // zero to one
  assert(fibonacci(0) == 0);
  assert(fibonacci(1) == 1);
  // two to four
  assert(fibonacci(2) == 1);
  assert(fibonacci(3) == 2);
  assert(fibonacci(4) == 3);
  // large
  assert(fibonacci(5) == 5);
  assert(fibonacci(12) == 144);
  // negative one to negative four
  assert(fibonacci(-1) == 1);
  assert(fibonacci(-2) == -1);
  assert(fibonacci(-3) == 2);
  assert(fibonacci(-4) == -3);
  // large negative
  assert(fibonacci(-5) == 5);
  assert(fibonacci(-8) == -21);

  return 0;
}
